### PR TITLE
Add ai methods that sort threads based on content similarity w/ given…

### PR DESCRIPTION
**Pull request by dmoralis**
# Description

Hi, I added a new feature to your project which refers to **sorting threads based on their similarity to the keywords given** by the user. The way this is achieved is through a HuggingFace library model, which creates an embedding from the title and content of each thread and compares it to the embedding of each keyword. Thus, for each thread the overall similarity to the keywords is summed up, through which the objects of type "Submission" are sorted and then given to the function "get_subreddit_undone" for further checks (allow_nsfw, already done etc). Finally in case that all the threads are used up then for each different date filter "VALID_TIME_FILTERS" applied the threads are re-sorted based on their similarity to the keywords. I am open to your opinion about this feature and sorry about any possible mistakes, this is my first contribution to a project.

# Issue Fixes

None

# Checklist:

- [x] I am pushing changes to the **develop** branch
- [x] I am using the recommended development environment
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have formatted and linted my code using python-black and pylint
- [x] I have cleaned up unnecessary files
- [x] My changes generate no new warnings
- [x] My changes follow the existing code-style
- [x] My changes are relevant to the project

How to test -> Run the main.py file

### CODE REVIEW

File: video_creation/data/videos.json [hunk 1]
```diff
@@ -1 +1,10 @@
...
+[
+    {
+        "subreddit": "news",
+        "id": "zo8im0",
+        "time": "1671295036",
+        "background_credit": "bbswitzer",
+        "reddit_title": "Anti-LGBTQ hate thrives online spurs fears of more violence",
+        "filename": "Anti-LGBTQ hate thrives online spurs fears of more violence.mp4"
+    }
+]
```
Comment (OpenSourceSimon) [inline video_creation/data/videos.json:10]:
```suggestion
[]
```

**Review by OpenSourceSimon — COMMENTED** [review]
Will need to test this but thanks! Good functionality!

**Review by OpenSourceSimon — APPROVED** [review]
Tested and works! Thank you for your PR

**Comment by dmoralis** [comment]
No problem :) Keep doing a great job

**Pull request merged**